src/com/amazon/ion/benchmark/IonSchemaUtilities.java

```diff
@@ -82,6 +87,8 @@ public static int parseConstraints(IonStruct value, String keyWord) throws IOExc
                             if (reader.getType() == IonType.SYMBOL) {
                                 if (reader.symbolValue().equals(KEYWORD_MAX)) {
                                     max = Integer.MAX_VALUE;
+                                } else if (keyWord.equals(IonSchemaUtilities.KEYWORD_TIMESTAMP_PRECISION)) {
+                                    max = Timestamp.Precision.valueOf(reader.stringValue().toUpperCase()).ordinal();
```

Comment:
What if the first element is an int and the second element is a timestamp? Should that be an error?

Reply:
> What if the first element is an int and the second element is a timestamp? Should that be an error?

The value processed by this method doesn't contain the combinations of int and timestamp. Normally this method will process the value like [min/int, max/int] ,[min/timestamp.precision, max/timestamp.precision], int. Correct me if I misunderstand this question.

Reply:
Well-formed Ion schema data won't have this problem. My question is: what happens if you receive an invalid schema? This question applies more broadly than just here. I want you to think about what kind of validation might be necessary. Or, is there a way to use an Ion Schema library to pre-validate the schema so that this doesn't need to be done in the data generator?

Reply:
> Well-formed Ion schema data won't have this problem. My question is: what happens if you receive an invalid schema? This question applies more broadly than just here. I want you to think about what kind of validation might be necessary. Or, is there a way to use an Ion Schema library to pre-validate the schema so that this doesn't need to be done in the data generator?

The method of checking validation of schema has been updated in the last commit. The method IonSchemaUtilities.checkValidationOfSchema will be called at the start place where ion schema file is loaded. This method is using the method `newSchema(inputSchema.iterator())` of ` ion schema kotlin` . If there's invalid schema syntax in the `inputSchema`, the `InvalidSchemaException` will be catched.

Reply:
Nice!